Design/reflectometry-idf.md

```diff
@@ -0,0 +1,62 @@
+# Reflectometry IDF
+
+## Motivation
+
+The reflectometry instrument scientists do not currently use an IDF and are heavily reliant on their own scripts.
+If the IDF could properly describe their setup then it would solve existing issues and improve their workflow.
+
+Potential benefits include, but are not limited to:
+  - Correct geometric description of detector components and calculation of angles to be fed into [algorithms](https://github.com/mantidproject/mantid/issues/26971)
+  - Complex beam paths to allow for accurate description of mirror action, with a view to accurately describe the effect of gravity
+  - Saving time by having multiple modes of configuration described within one file
+  - Reduce known sources of error due to problems with Q error bars, incorrect angles, gravity corrections, etc.
+  - Facilitate new equipment over multiple instruments/facilities, such as 2D detectors
+
+## Solution
+
+This task should be thought of in terms of four main sub-tasks:
+
+  ### 1. POLREF IDF support for relative pixel map
+  
+  Andrew currently has a scan of the POLREF detector which he uses in a script to move each detector pixel into place every time 
```

Comment:
Can you elaborate on this scanning technique. What exactly are the steps to this? I think you're saying that they use the specular reflection (along with known incident angle) to identify exactly where the detector is? Do they use [this](https://docs.mantidproject.org/nightly/algorithms/FindReflectometryLines-v2.html) for this or something else?

Reply:
I don't have any further information about the scan other than that it is of theta vs channel, and was necessary because they don't know where each pixel is. But yes, from what I can tell they use specular reflection along with a known incident angle to work backwards from the expected reflected angle and move the detector into place (with the relative pixel positions described by the scan/relative pixel map).

Reply:
It is necessary for you to confirm with the team that that is what they are doing. I.e some simple trig based on incident angle and known sample-detector distances.

Reply:
Hi Owen, The scan we are referring to is a detector map, where we work out the position of each pixel in Theta (degrees). We do this by putting a supermirror at the sample point to bounce the beam up. This is just so we can used the bench to map the whole detector as if you use the straight through beam then the detector bench does not drive low enough. We then start with the detector as low as possible with the beam off the top of the detector. The bench is then moved up in very small steps in theta. I calculated a minimum step size via the special Nyquist frequency. We then begin and end a run from some small number of micro amps (we actually used event mode with a high frequency log of the theta angle and sliced it all up later). Then we move by the next small step and do another TOF scan and so on, until we sweep of the bottom of the detector. Once we have this scan, we then go through it for each pixel and obtain as a function of angle the intensity on that pixel. This is effectively a theta scan for that pixel; we fit it and obtain for pixel 1 say its position in theta. We do this for all 640 pixels on the detector. we define 0 theta as pixel 280, but that does not really matter, as its the relative distances between the pixels that’s important, The way the detectors are made means that the pixels are not evenly spaced and this allows use to obtain a pixel position and error on it form the fit. We can use this then to remap the detector map to actual theta positions. This means the data converts to Q correctly for 2D maps. It also allows proper implementation of Q summing, gravity etc. We cannot just use an IDF detector description where we say each pixel is just 0.5mm apart as we get the wrong answer when converting to Q as each pixel is uniquely positioned, because of the manufacturing process. We just load this pixel vs theta data set in (2-column dataset) and use it to calculate relative pixel positions. 

I hope this helps, please ask my or Andrew any questions you have.